src/memkind.c

```diff
@@ -582,6 +582,9 @@ void memkind_init(memkind_t kind, bool check_numa)
             abort();
         }
     }
+    void *t = memkind_malloc(kind, 8);
```

Comment:
What if it fails? E.g. DAX_KMEM returns null if no pmem is detected in the system, in such case, we would get segfault